Drone: Download only required Boost.CI scripts on Windows

Flamefire opened the pull request:
Follow-up to #197

Test build using modified drone.star: https://drone.cpp.al/boostorg/boost-ci/374

sdarwin commented:
- Tested, and hit "Retry not supported".  The way to solve that (already mentioned in functions.star) is to preface the first download command with `pwsh.exe -Command` because `pwsh.sh` is a newer version of powershell that supports retries.  

- When I was first adding that feature with pwsh, I also put that same text "echo "Retry not supported" which is very helpful for debugging. But after being satisfied that everything was working properly, I ended up removing that line to reduce log output because Drone echoes everything.

Flamefire commented:
> * Tested, and hit "Retry not supported".  The way to solve that (already mentioned in functions.star) is to preface the first download command with `pwsh.exe -Command` because `pwsh.exe` is a newer version of powershell that supports retries.

Ah that was the reason. So there are 2 PowerShells in the image? Why is the older one used anyways?

>     * When I was first adding that feature with pwsh, I also put that same text "echo "Retry not supported" which is very helpful for debugging. But after being satisfied that everything was working properly, I ended up removing that line to reduce log output because Drone echoes everything.

So that's the reason. I was confused about the `echo 'Using powershell'` at https://github.com/boostorg/boost-ci/blob/70b65ca7c37c488836794d13806e5afdffd47bdc/ci/drone/functions.star#L154 in the part which does **not** call `pwsh`

If the images stay then we can use the now fixed version. Test build with actual downloads: https://drone.cpp.al/boostorg/boost-ci/379 https://drone.cpp.al/boostorg/boost-ci/380
They fail later as expected because I deleted the whole ci folder and as the repo is boost-ci it doesn't download it again in ´boost.*`

sdarwin commented:
windows uses the older version of powershell as the default, if you'd like to run the newer version, then install it and call `pwsh.exe`.

Flamefire commented:
> windows uses the older version of powershell as the default, if you'd like to run the newer version, then install it and call pwsh.exe.

What do you mean by "install it"? This drone script doesn't install anything but still has the newer powershell when `pwsh.exe` is called, doesn't it? So there must be 2 powershells, or am I missing something?

sdarwin commented:
Sorry, that was a general comment about how it might be used.  Specifically with Drone it's installed by the Dockerfile, and so already available.  Yes "2 powershells".
